terraform/consul/README.md

```diff
@@ -0,0 +1,139 @@
+--------------------------------------------------------------------------------
+consul-client-a   running   8300-8302/tcp, 8500/tcp, 8301-8302/udp, 8600/tcp, 8600/udp
+consul-client-b   running   8300-8302/tcp, 8301-8302/udp, 8600/tcp, 8600/udp, 0.0.0.0:8501->8500/tcp
+```
+
+You can also visit the Consul UI at http://localhost:8500/ui and verify that the services have been registered.
+
+### Step 4: Deploy the services
+
+Now that the infrastructure is up and running we can deploy the services. To do this we will be using a helper script in `bin/servicesctl` which provides a utility wrapper around the Consul HTTP API making interactions simpler. The script will read the service definitions from the `api-service.json` and `web-service.json` files and register them with Consul.
+
+```bash
+./bin/servicesctl deploy
+```
+
+You can verify that the services have been registered by visiting the Consul UI at http://localhost:8500/ui and clicking on the `Services` tab.
+
+### Step 5: Verify the DNS records are created
+
+Once the services have been registered you can verify that the DNS records have been created by the CTS container. You should be able to see the logs from the CTS container by running the following command:
+
+```bash
+docker logs cts
+```
+
+You should also be able to see the changes at DNSimple. If you had used a production account with a real domain you would be able to verify the changes by running the following command:
+
+```bash
+dig @ns1.dnsimple.com api.example.com
+```
+
+### Step 6: Update the services
+
+Now that the services have been deployed we can update the service definitions and see how the CTS container reacts to the changes. To do this we will be updating the `api-service.json` and `web-service.json` files and changing the `port` value to a value of your choice. Once you have updated the files you can run the following command to update the services:
+
+```bash
+./bin/servicesctl deploy
+```
+
+You should be able to see the changes in the logs from the CTS container as well as at DNSimple. If you had used a production account with a real domain you would be able to verify the changes through a DNS lookup via something like `dig`.
+
+### Step 7: Cleanup
+
+Once you are done with the demo you can revert the changes by running the following command:
+
+```bash
+./bin/servicesctl teardown
+```
+
+This will remove the services from Consul and also remove the DNS records from DNSimple as the CTS container reacts to the changes.
+
+Next, you can stop the containers by running the following command:
+
+```bash
+docker compose down
+```
+
+### Troubleshooting
+
+- If you are having issues with the CTS container not applying the changes and you do not see any errors in the logs you can change the log level to `DEBUG` in the `config/cts-config.hcl` file and restart the container. This will provide more verbose logging which should help you debug the issue. To restart the container you can run the following command:
```

Comment:
```suggestion
- If you are having issues with the CTS container not applying the changes, and you do not see any errors in the logs, change the log level to `DEBUG` in the `config/cts-config.hcl` file, and restart the container. This will provide more verbose logging, which should help you debug the issue. To restart the container, run the following command:
```